pennylane_cirq/simulator_device.py

```diff
@@ -71,225 +59,115 @@ class SimulatorDevice(CirqDevice):
-            basis_state_idx = np.sum(2 ** np.argwhere(np.flip(basis_state_array) == 1))
-            self.initial_state[basis_state_idx] = 1.0
-
-        elif operation == "QubitStateVector":
-            if not self._first_apply:
-                raise qml.DeviceError(
-                    "The operation QubitStateVector is only supported at the beginning of a circuit."
-                )
+        self._initial_state = np.zeros(2 ** len(self.qubits), dtype=np.complex64)
+        basis_state_idx = np.sum(2 ** np.argwhere(np.flip(basis_state_array) == 1))
+        self._initial_state[basis_state_idx] = 1.0
 
-            if not self.analytic:
-                raise qml.DeviceError(
-                    "The operation QubitStateVector is only supported in analytic mode."
-                )
+    def apply_qubit_state_vector(self, qubit_state_vector_operation):
+        if not self.analytic:
+            raise qml.DeviceError(
+                "The operation QubitStateVector is only supported in analytic mode."
+            )
 
-            state_vector = np.array(par[0], dtype=np.complex64)
+        state_vector = np.array(
+            qubit_state_vector_operation.parameters[0], dtype=np.complex64
+        )
 
-            if len(state_vector) != 2 ** len(self.qubits):
-                raise qml.DeviceError(
-                    "For QubitStateVector, the state has to be specified for the correct number of qubits. Got a state of length {}, expected {}.".format(
-                        len(state_vector), 2 ** len(self.qubits)
-                    )
+        if len(state_vector) != 2 ** len(self.qubits):
+            raise qml.DeviceError(
+                "For QubitStateVector, the state has to be specified for the correct number of qubits. Got a state of length {}, expected {}.".format(
+                    len(state_vector), 2 ** len(self.qubits)
                 )
+            )
 
-            norm_squared = np.sum(np.abs(state_vector) ** 2)
-            if not np.isclose(norm_squared, 1.0, atol=1e-3, rtol=0):
-                raise qml.DeviceError(
-                    "The given state for QubitStateVector is not properly normalized to 1.0. Got norm {}".format(
-                        math.sqrt(norm_squared)
-                    )
+        norm_squared = np.sum(np.abs(state_vector) ** 2)
+        if not np.isclose(norm_squared, 1.0, atol=1e-3, rtol=0):
+            raise qml.DeviceError(
+                "The given state for QubitStateVector is not properly normalized to 1.0. Got norm {}".format(
+                    math.sqrt(norm_squared)
                 )
+            )
 
-            self.initial_state = state_vector
-
-        if self._first_apply:
-            self._first_apply = False
+        self._initial_state = state_vector
 
-    def pre_measure(self):
-        super().pre_measure()
+    def apply(self, operations, rotations=None, **kwargs):
+        super().apply(operations, rotations, **kwargs)
```

Comment:
```suggestion
        super().apply(operations, rotations=rotations, **kwargs)
```

Reply:
This would be in case `kwargs=={}` is `True` and `apply` being called only with passing `operations` to it, since then I think `rotations` would default to `None` and there might be an error arising from passing too many positional arguments (`None` is going to be passed as the second argument). Might be wrong though.

Reply:
I removed the argument `rotations` entirely now to be consistent with the signature of `apply` from the `QubitDevice` class.